lib/phoenix_live_view/test/dom.ex

```diff
@@ -426,80 +399,6 @@ defmodule Phoenix.LiveViewTest.DOM do
       end)
 
     cond do
-      # reorder and/or remove stream children
-      content_changed? && type == "stream" ->
-        children = updated_existing_children ++ updated_appended
-
-        # we always reduce over all streams, even if only one of them is actually relevant
-        # for this container...
-        new_children =
-          Enum.reduce(streams, updated_existing_children, fn item, acc ->
-            [ref, inserts, _deletes | maybe_reset] = item
-            reset = maybe_reset == [true]
-
-            # remove all children that are in a stream that was reset
-            acc =
-              if reset do
-                Enum.reject(acc, fn child ->
-                  item_ref = attribute(child, "data-phx-stream")
-                  item_ref == ref
-                end)
-              else
-                acc
-              end
-
-            Enum.reduce(inserts, acc, fn [id, insert_at, limit], acc ->
-              old_index = Enum.find_index(children, &(attribute(&1, "id") == id))
-              current_index = Enum.find_index(acc, &(attribute(&1, "id") == id))
-
-              appended? = Enum.any?(updated_appended, &(attribute(&1, "id") == id))
-
-              existing? = Enum.any?(updated_existing_children, &(attribute(&1, "id") == id))
-              deleted? = MapSet.member?(stream_deletes, id)
-
-              child =
-                case old_index && Enum.at(children, old_index) do
-                  nil -> nil
-                  child -> set_attr(child, "data-phx-stream", ref)
-                end
-
-              parent_id = parent_id(html_tree, id)
-
-              cond do
-                !child ->
-                  acc
-
-                # skip added children that aren't ours if they are not being appended
```

Comment:
To be discussed: how are we handling non-stream items inside a stream container? I don't think this is something that we officially support, right? I that case we could also check for non stream elements and warn users.

Reply:
Should we raise ? so we force people to fix both for the benefit of both JS and Elixir clients?

Reply:
Something like this maybe: https://github.com/phoenixframework/phoenix_live_view/pull/3078/commits/a4e0b8024fbd6db96fd42ff28d43020717dbd1c0

<img width="670" alt="image" src="https://github.com/phoenixframework/phoenix_live_view/assets/4116351/5cd8945e-afc8-4480-8318-7691aadd14d8">

Reply:
Confirm we do not support non stream items in a stream container so this error looks good!